sklearn/utils/validation.py

```diff
@@ -1314,6 +1319,11 @@ def _check_sample_weight(sample_weight, X, dtype=None):
         if sample_weight.shape != (n_samples,):
             raise ValueError("sample_weight.shape == {}, expected {}!"
                              .format(sample_weight.shape, (n_samples,)))
+
+    if sample_weight is not None and ensure_nonnegative:
+        if not np.all(sample_weight >= 0):
+            raise ValueError("Sample weights must be non-negative")
```

Comment:
Checking for negative sample weight is a separate issue (see discussion in https://github.com/scikit-learn/scikit-learn/issues/15531 and linked PRs). There are cases where negative sample weight make sense apparently..  So we should not do this check as part of this PR.

Reply:
This code was already implemented by the superseded PR. I can remove it.